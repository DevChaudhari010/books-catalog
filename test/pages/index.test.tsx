import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "@pages/index";

describe("Home page", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });
  test("Check for Books tile", () => {
    const { getByText } = render(<Home />);
    expect(getByText("Books")).toBeInTheDocument();
  });
  test("Check for Technology tile", () => {
    const { getByText } = render(<Home />);
    expect(getByText("Technology Choices")).toBeInTheDocument();
  });
});
