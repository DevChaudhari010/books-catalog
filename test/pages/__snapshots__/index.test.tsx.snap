// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid"
  >
    <a
      class="card"
      href="/books"
    >
      <h2>
        Books
        <span>
           →
        </span>
      </h2>
      <p>
        Browse Books we have!!.
      </p>
    </a>
    <a
      class="card"
      href="/technology"
    >
      <h2>
        Technology Choices
        <span>
           →
        </span>
      </h2>
      <p>
        Find information about techonlogies used in this project.
      </p>
    </a>
  </div>
</DocumentFragment>
`;
